src/lang-ietfabnf.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright (C) 2008 Julian Reschke
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * Registers a language handler for IETF ABNF, as specified in RFC 5234,
+ * and extended in RFCs 7230 and 7405.
+ */
+
+PR['registerLangHandler'](
+    PR['createSimpleLexer'](
+        [
+         // comment
+         [PR['PR_COMMENT'], /;[^\r\n]*/, null, ";"],
+        ],
+        [
+         // string, binary, decimal and hex literals
+         [PR['PR_LITERAL'], /((\%s)?"[^"]*"|(\%x[A-Za-z0-9]+(-[A-Za-z0-9]+|[\.[A-Za-z0-9]+]*))|(\%d[0-9]+(-[0-9]+|[\.[0-9]+]*))|(\%b[01]+(-[01]+|[\.[01]+]*)))/, null],
```

Comment:
I'm pretty sure `(-[A-Za-z0-9]+|[\.[A-Za-z0-9]+]*)` doesn't do what you want.  Notice the `[` and `]` around `\.[A-Za-z0-9]+` instead of parentheses.

Maybe `(?:[-.][A-Za-z0-9]+)*` does what you want.

If so, this problem occurs thrice.

Also, AFAICT, `%s` doesn't appear in RFC 5234.  Is that part of an extension?  Do those conventions include an escaping convention for strings?

Reply:
1) Will check.
2) %s is defined in <https://tools.ietf.org/html/rfc7405#section-2.1> (but so is "%i" which I should include as well)

Reply:
I tried to address this in https://github.com/google/code-prettify/pull/542/commits/d8a2671d424142f07dec1c64f6d4cb8cf858a607 - note that the simplified expression would match things like

   %x31-39-49

which it should not (but then one could argue that it's not the job of the highlighter to spot errors like these).